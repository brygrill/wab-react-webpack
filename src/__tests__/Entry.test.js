import React from 'react';
import ReactDOM from 'react-dom';
import toJson from 'enzyme-to-json';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import MyComponent from '../components/Entry';

configure({ adapter: new Adapter() });

const mockObj = {};

describe('<MyComponent />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MyComponent wab={mockObj} esriJS={mockObj} />, div);
  });

  it('initially shows loader', () => {
    const component = shallow(<MyComponent wab={mockObj} esriJS={mockObj} />);

    component.setState({
      loading: true,
    });

    expect(toJson(component)).toMatchSnapshot();
  });

  it('shows children after load', () => {
    const component = shallow(<MyComponent wab={mockObj} esriJS={mockObj} />);

    component.setState({
      loading: false,
    });

    expect(toJson(component)).toMatchSnapshot();
  });
});
