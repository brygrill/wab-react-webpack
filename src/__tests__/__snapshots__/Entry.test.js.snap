// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyComponent /> initially shows loader 1`] = `
<Loading
  content="Loading"
/>
`;

exports[`<MyComponent /> shows children after load 1`] = `<ErrorMessage />`;
